server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -31,15 +31,42 @@
 
     private static final Object mutex = new Object();
 
+    private static final StackTraceElement[] EMPTY = new StackTraceElement[0];
     private static final DateFormatter DATE_TIME_FORMATTER = DateFormatter.forPattern("date_optional_time");
 
     private int busiestThreads = 3;
     private TimeValue interval = new TimeValue(500, TimeUnit.MILLISECONDS);
-    private TimeValue threadElementsSnapshotDelay = new TimeValue(10);
+    private TimeValue threadElementsSnapshotDelay = new TimeValue(10, TimeUnit.MILLISECONDS);
     private int threadElementsSnapshotCount = 10;
-    private String type = "cpu";
+    private ReportType type = ReportType.CPU;
     private boolean ignoreIdleThreads = true;
 
+    public enum ReportType {
+
+        CPU("cpu"),
+        WAIT("wait"),
+        BLOCK("block");
+
+        private final String type;
+
+        ReportType(String type) {
+            this.type = type;
+        }
+
+        public String getTypeValue() {
+            return type;
+        }
+
+        public static ReportType of(String type) {
```

Comment:
Don't all enums have `valueOf` that we can use?

Reply:
The only reason I made a custom 'valueOf' is because I wanted to throw a nicer message in case the enum type doesn't exist. Enum.valueOf will say something like 'no enum constant...' with internal information about the class, which will be confusing to the users. It's a static method, so we can't override it.